### PR TITLE
[WIP][v2] Transfer logic of devfile support check from odo

### PR DESCRIPTION
**Pull request by maysunfaisal**
Signed-off-by: Maysun J Faisal <maysun.j.faisal@ibm.com>

For Issue https://github.com/openshift/odo/issues/3249
This PR moved the logic from odo to this repo to check if a devfile is supported.

This is done so that odo does not parse the devfile but rather read the index.json for information and this improves `odo catalog list components` performance with big registries

### CODE REVIEW

File: tools/go.mod [hunk 1]
```diff
@@ -1,5 +1,8 @@
-module github.com/elsony/devfile-registry/tools
+module github.com/odo-devfiles/registry/tools
 
 go 1.14
 
-require gopkg.in/yaml.v2 v2.2.8
+require (
+	github.com/ghodss/yaml v1.0.0
```
Comment (johnmcollier) [inline tools/go.mod]: Is there a reason why we can't use the yaml package that's already pulled in?

File: tools/utils/utils.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+)
+
+// IsDevfileSupported checks if devfile v2 is supported
+func IsDevfileSupported(devfile types.Devfile) bool {
+
+	hasComponentContainer := false
+	hasComponentContainerName := false
+	hasRunGroupCommand := false
+
+	for _, component := range devfile.Components {
```
Comment (johnmcollier) [inline tools/utils/utils.go:18]: Could we simplify these two for loops a bit? Maybe something like:

```
hasSupportedContainer := false
hasRunGroupCommand := false
for _, component := range devfile.Components {
    if component.Container != nil && component.Container.Name != "" {
        hasSupportedContainer = true
        break
    }
}

for _, command := range devfile.Commands {
    if hasRunGroupCommand = command.Exec != nil && command.Exec.Group != nil && command.Exec.Group.Kind == runGroup {
       hasRunGroupCommand = true
       break
    }
}

return hasSupportedContainer && hasRunGroupCommand
```

**Review by johnmcollier — CHANGES_REQUESTED** [review]

File: tools/utils/utils.go (hunk 2 above)
Comment (johnmcollier) [reply tools/utils/utils.go:18]: And maybe it makes sense to separate out each for loop into a function?

Comment (maysunfaisal) [reply tools/utils/utils.go:18]: Updated with feedback, I had just copy pasted it from odo repo before. It looks cleaner now. Changes also made to v1 PR.

File: tools/go.mod (hunk 1 above)
Comment (maysunfaisal) [reply tools/go.mod]: it was an auto-import after copy pasting code from odo, fixed it now. Changes also made to v1 PR.

**Comment by maysunfaisal** [comment]
Closing this PR.

We dont want to show the supported column on odo catalog list components. The devfile v2 registry already has the name field.

See discussion https://github.com/openshift/odo/issues/3249#issuecomment-641350713

**Pull request closed without merging**